services/identity/service.go

```diff
@@ -281,33 +292,61 @@ func (s *Service) Propagate(msg network.Body) {
 
 // getIdentityStorage returns the corresponding IdentityStorage or nil
 // if none was found
-func (s *Service) getIdentityStorage(id ID) *storage {
+func (s *Service) getIdentityStorage(id ID) *Storage {
 	s.identitiesMutex.Lock()
 	defer s.identitiesMutex.Unlock()
-	is, ok := s.identities[string(id)]
+	is, ok := s.Identities[string(id)]
 	if !ok {
 		return nil
 	}
 	return is
 }
 
 // setIdentityStorage saves an IdentityStorage
-func (s *Service) setIdentityStorage(id ID, is *storage) {
+func (s *Service) setIdentityStorage(id ID, is *Storage) {
 	s.identitiesMutex.Lock()
 	defer s.identitiesMutex.Unlock()
 	log.Lvlf3("%s %x %v", s.Context.ServerIdentity(), id[0:8], is.Latest.Device)
-	s.identities[string(id)] = is
+	s.Identities[string(id)] = is
+}
+
+// saves the actual identity
+func (s *Service) save() {
+	log.Lvl3("Saving service")
+	b, err := network.MarshalRegisteredType(s)
+	if err != nil {
+		log.Error("Couldn't marshal service:", err)
+	} else {
+		err = ioutil.WriteFile(s.path+"/identity.bin", b, 0660)
+		if err != nil {
+			log.Error("Couldn't save file:", err)
+		}
+	}
 }
 
 func newIdentityService(c *sda.Context, path string) sda.Service {
 	s := &Service{
 		ServiceProcessor: sda.NewServiceProcessor(c),
-		identities:       make(map[string]*storage),
+		Identities:       make(map[string]*Storage),
 		skipchain:        skipchain.NewClient(),
 		path:             path,
 	}
+	b, err := ioutil.ReadFile(path + "/identity.bin")
+	if err != nil && !os.IsNotExist(err) {
+		log.Error("Error while reading", path+"/identity.bin:", err)
```

Comment:
Yeah but when creating a `Service` with a given data and that fails, that means the returned `Service` is not initialized as expected. Again, that's the same question: should we print & continue or return an error ...
I'm not sure either but it's a question we need to solve .!